app/src/androidTest/kotlin/at/bitfire/davdroid/db/AppDatabaseMigrationsTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.db
+
+import android.content.Context
+import androidx.room.Room
+import androidx.room.testing.MigrationTestHelper
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
+import androidx.test.platform.app.InstrumentationRegistry
+import at.bitfire.davdroid.db.Collection.Companion.TYPE_CALENDAR
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+
+class AppDatabaseMigrationsTest {
+
+    private val TEST_DB = "test"
+
+    val context: Context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    private val autoMigrationSpecs = AppDatabase.autoMigrationSpecs.map { it(context) }
+
+    @Rule
+    @JvmField
+    val helper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        AppDatabase::class.java,
+        autoMigrationSpecs,
+        FrameworkSQLiteOpenHelperFactory()
+    )
+
+    /**
+     * Used for testing the migration process between two versions.
+     * @param fromVersion The version from which to start testing.
+     * @param toVersion The target version to test
+     * @param setup Run here all the SQL queries that prepare the database with the data required
+     * to perform the migration process.
+     * @param block Run here all the assertions for [AppDatabase] to make sure the migration was
+     * performed correctly.
+     */
+    private fun testMigration(
+        fromVersion: Int,
+        toVersion: Int,
+        setup: SupportSQLiteDatabase.() -> Unit,
+        block: (AppDatabase) -> Unit
+    ) {
+        var db = helper.createDatabase(TEST_DB, fromVersion).apply {
+            setup()
+
+            // Prepare for the next version.
+            close()
+        }
+
+        // Re-open the database with the new version and provide all the migrations.
+        db = helper.runMigrationsAndValidate(TEST_DB, toVersion, true, *AppDatabase.migrations)
+
+        // Once the migration process is complete, load AppDatabase
+        val database = Room.databaseBuilder(context, AppDatabase::class.java, TEST_DB)
+            // Include all manual migrations
+            .addMigrations(*AppDatabase.migrations)
+            // And all auto migration specs
+            .apply {
+                for (spec in autoMigrationSpecs) {
+                    addAutoMigrationSpec(spec)
+                }
+            }
+            .build()
+        block(database)
+    }
+
+    /**
+     * Test migrations from full VTIMEZONE to just timezone ID
+     * @see <a href="https://github.com/bitfireAT/davx5-ose/pull/1104">PR</a>
```

Comment:
I don't see any reason not to have links to PR requests in the kdoc, but it does seem a bit odd to me. Not sure we want that to become a general practice or habit ... I guess as soon as another PR changes the code we'd have to update it again and there's the risk of forgetting to do that and in my head it could become messy really quick.

Maybe it's ok only for the migration tests ... idk.

While `@see` can be used for hyperlinks it's [intended to give the reader a better understanding of relationships](https://kotlinlang.org/docs/api-guidelines-informative-documentation.html#use-explicit-links-in-documentation) and is not meant to provide additional information. 

If we really really want to add the link to the PR it should be added as an [external link](https://kotlinlang.org/docs/kotlin-doc.html#external-links) instead.

Reply:
Sounds fair, then do I remove it completely or add it as a link? What do you think @rfc2822 ?

Reply:
Just add it as a link for now. I do see the appeal of jumping straight to the PR in case we need to work on a migration later on. Ricki can remove it when they do the final review after me.